src/hazelcore/Action_Pat.re

```diff
@@ -1255,7 +1381,45 @@ and ana_perform_operand =
         Succeeded((zp, ctx, u_gen));
       };
     }
-
+  | (_, TypeAnnZP(_, zop, ann)) =>
+    switch (ana_perform_operand(ctx, u_gen, a, zop, ty)) {
+    | Failed => Failed
+    | CursorEscaped(side) =>
+      ana_perform_operand(
+        ctx,
+        u_gen,
+        Action_common.escape(side),
+        zoperand,
+        ty,
+      )
+    | Succeeded((ZOpSeq(_, zseq), ctx, u_gen)) =>
+      let newseq = annotate_last_operand(zseq, ann);
```

Comment:
a comment explaining what's going on here might be useful (have to reassociate the type annotation since its not parsed as part of the opseq system)